workflow/executor/retry/retry.go

```diff
@@ -22,3 +22,32 @@ var (
 	Jitter        = env.LookupEnvFloatOr("EXECUTOR_RETRY_BACKOFF_JITTER", 0.5)
 	ExecutorRetry = wait.Backoff{Steps: Steps, Duration: Duration, Factor: Factor, Jitter: Jitter}
 )
+
+// GetExecutorRetry returns a wait.Backoff configuration for retries,
+// with values determined in the following order of precedence:
+//
+//   1. If the corresponding environment variable is set, its value is used.
+//      - EXECUTOR_RETRY_BACKOFF_STEPS
+//      - EXECUTOR_RETRY_BACKOFF_DURATION
+//      - EXECUTOR_RETRY_BACKOFF_FACTOR
+//      - EXECUTOR_RETRY_BACKOFF_JITTER
+//
+//   2. If the env var is not set, the value provided via function arguments is used.
+//
+// Parameters:
+//   - steps: fallback number of retry steps.
+//   - duration: fallback base duration between retries.
+//   - factor: fallback exponential backoff factor.
+//   - jitter: fallback jitter factor for randomness.
+//
+// Returns:
+//   - wait.Backoff: configured backoff parameters.
+func GetExecutorRetry(duration time.Duration, factor, jitter float64, steps int, cap time.Duration) wait.Backoff {
+	return wait.Backoff{
+		Steps:    env.LookupEnvIntOr("EXECUTOR_RETRY_BACKOFF_STEPS", steps),
+		Duration: env.LookupEnvDurationOr("EXECUTOR_RETRY_BACKOFF_DURATION", duration),
+		Factor:   env.LookupEnvFloatOr("EXECUTOR_RETRY_BACKOFF_FACTOR", factor),
+		Jitter:   env.LookupEnvFloatOr("EXECUTOR_RETRY_BACKOFF_JITTER", jitter),
+		Cap: cap,
```

Comment:
Perhaps we can use `EXECUTOR_RETRY_BACKOFF_CAP`,
since we are providing a default `cap` we want the ability to overwrite this